ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -387,6 +387,10 @@ def _log(self, sql):
     def _get_sqla_table(
         self, name: str, schema: str | None = None, autoload: bool = True, **kwargs: Any
     ) -> sa.Table:
+        # If the underlying table (or more likely, view) has changed, remove it
+        # to ensure a correct reflection
+        if autoload and self.inspector.has_table(name):
+            self.meta.remove(sa.Table(name, self.meta))
```

Comment:
Is the coarseness issue that it may be expensive to repeatedly call `con.table("t")`?

Reply:
That was my thinking, that calling tables now involves this extra reflection.

We _could_ probably do a set comparison of the existing columns in `self.meta` and the columns in `self._metadata` but I don't know if that will be more expensive

Reply:
Eh, I don't think it's worth it yet.

Some very light profiling against the snowflake backend (one of the slowest backends for metadata queries) shows about a 20% performance hit:

```
# `master`
In [10]: %timeit con.table("diamonds")
1.41 s ± 223 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

# this PR
In [4]: %timeit con.table("diamonds")
1.75 s ± 229 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

Since this is a correctness issue I think we should fix it even if there's a performance hit (for now).